### PR TITLE
task: add unit tests, accessibility fixes for interstitial screen web component

### LINKED ISSUE #7902: InterstitialScreen: Add unit tests, examples and update doc for InterstitialScreen webcomponent
- Add unit tests
- Create stackblitz examples
- Update docs section
- Keyboard and accessibility Fixes

### PR DESCRIPTION
**Pull request by amal-k-joy**
Closes #7902 

- Added unit tests
- Make inactive steps in accessible using inert attribute and aria-hidden
- Retain focus on action buttons during step navigation edge cases, like when next button is removed in last step and start button is displayed, it should automatically transfer focus to start button
- Added arial live announcement for screen readers using step changes

#### How did you test and verify your work?
yarn test coverage, storybook, screenreader
#### PR Checklist

As the author of this PR, before marking ready for review, confirm you:

- [ ] Reviewed every line of the diff
- [ ] Updated documentation and storybook examples
- [ ] Wrote passing tests that cover this change
- [ ] Addressed any impact on accessibility (a11y)
- [ ] Tested for cross-browser consistency
- [ ] Validated that this code is ready for review and status checks should pass

More details can be found in the [pull request](./CONTRIBUTING.md) section of
our contributing docs.

### CODE REVIEW

File: packages/ibm-products-web-components/src/components/interstitial-screen/interstitial-screen-body.ts [hunk 1]
```diff
@@ -31,20 +31,23 @@ const blockClass = `${prefix}--interstitial-screen`;
...
 
   private carouselAPI!: InitCarousel;
   private carouselElement = createRef<HTMLElement>();
 
   firstUpdated(): void {
-    const slot = this.shadowRoot?.querySelector('slot') as HTMLSlotElement;
-
-    const assigned = slot.assignedElements({ flatten: true });
+    const bodyItems = this.querySelectorAll(
+      'c4p-interstitial-screen-body-item'
```
Comment (matthewgallo) [inline packages/ibm-products-web-components/src/components/interstitial-screen/interstitial-screen-body.ts]: Use package prefix

File: packages/ibm-products-web-components/src/components/interstitial-screen/interstitial-screen-body.ts [hunk 2]
```diff
@@ -107,6 +116,27 @@ class CDSInterstitialScreenBody extends HostListenerMixin(LitElement) {
...
+
+    Object.values(allViews)?.forEach((item, idx) => {
+      const isActive = idx === itemNumber;
+
+      if (item) {
+        // Set aria-hidden based on active state
+        item.setAttribute('aria-hidden', String(!isActive));
+
+        if (!isActive) {
+          item.setAttribute('inert', ''); // Disable interactivity
```
Comment (matthewgallo) [inline packages/ibm-products-web-components/src/components/interstitial-screen/interstitial-screen-body.ts:130]: I like the usage of `inert` here 👏

File: packages/ibm-products-web-components/src/utilities/carousel/carousel.ts [hunk 3]
```diff
@@ -238,26 +326,67 @@ export const initCarousel = (
...
 
+  /**
+   * Removes event listeners for 'animationend' and 'transitionend' events from all elements with references stored in the `refs` object.
+   * Also registers swipe events if `excludeSwipeSupport` is false.
+   */
   const destroyEvents = () => {
-    registerSwipeEvents(carouselContainer, navigateNext, navigatePrev, true);
-  };
+    Object.values(refs).forEach((el) => {
+      console.log(el);
```
Comment (matthewgallo) [inline packages/ibm-products-web-components/src/utilities/carousel/carousel.ts]: Remove console log

File: packages/ibm-products-web-components/src/utilities/carousel/carousel.ts [hunk 4]
```diff
@@ -283,6 +412,7 @@ export const initCarousel = (
     reset,
     goToIndex,
     getActiveItem,
-    destroyEvents: !excludeSwipeSupport ? destroyEvents : null,
+    destroyEvents: destroyEvents,
```
Comment (matthewgallo) [inline packages/ibm-products-web-components/src/utilities/carousel/carousel.ts]:
```suggestion
    destroyEvents,
```

File: packages/ibm-products-web-components/src/components/interstitial-screen/interstitial-screen-header.ts [hunk 5]
```diff
@@ -92,6 +93,17 @@ class CDSInterstitialScreenHeader extends SignalWatcher(
...
 
+  getElementForAriaLive = () => {
+    const currentStep = interstitialDetailsSignal.get().currentStep;
+    const stepDetails = interstitialDetailsSignal.get().stepDetails;
+    return html` <div
+      aria-live="polite"
+      aria-atomic="true"
+      class="${carbonPrefix}--visually-hidden"
+    >
+      Step ${currentStep + 1} of ${stepDetails.length}
```
Comment (matthewgallo) [inline packages/ibm-products-web-components/src/components/interstitial-screen/interstitial-screen-header.ts:104]: We might want to think about if there's a better way for globalization support here

Comment (amal-k-joy) [reply packages/ibm-products-web-components/src/components/interstitial-screen/interstitial-screen-header.ts:104]: Yes.. thats a good point to consider
